lib/android.js

```diff
@@ -203,17 +205,33 @@ exports.detect = function detect(config, opts, finished) {
 							queue.push(function (cb) {
 								findNDK(subdir, config, cb);
 							});
-							// Android Studio installs the NDK to ~/Library/Android/Sdk/ndk-bundle
-							fs.statSync(subdir).isDirectory() && fs.readdirSync(subdir).forEach(function (name) {
-								if (/ndk-bundle/i.test(name)) {
+
+							// Check under NDK side-by-side directory which contains multiple NDK installations.
+							// Each subfolder is named after the version of NDK installed under it. Favor newest version.
+							const ndkSideBySidePath = path.join(subdir, 'ndk');
+							if (fs.existsSync(ndkSideBySidePath) && fs.statSync(ndkSideBySidePath).isDirectory()) {
+								const fileNames = fs.readdirSync(ndkSideBySidePath);
+								fileNames.sort(createReverseComparerFrom(versionStringComparer));
```

Comment:
I think this could just be a simple sort:
```JS
fileNames.sort((a, b) => {
    if (a > b) {
        return -1;
    } else if (b > a) {
        return 1;
    }
    return 0;
});
```

It seems to satisfy all possible cases:
```JS
const VERSIONS = [
    '20.1.5948944',
    '21.0.6113670',
    '.',
    '21.0.6113669',
    '16.1.44',
    '18.0',
    '19',
    '..',
    '0.1',
    '23',
    '20.2.0000001-alpha'
];
const SORTED = VERSIONS.sort((a, b) => {
    if (a > b) {
        return -1;
    } else if (b > a) {
        return 1;
    }
    return 0;
});
```
```
[
    "23",
    "21.0.6113670",
    "21.0.6113669",
    "20.2.0000001-alpha",
    "20.1.5948944",
    "19",
    "18.0",
    "16.1.44",
    "0.1",
    "..",
    "."
]
```

Reply:
A string compare won't work correctly for something like this...
`9.0`
`10.0`

...because ASCII character `1` is less than `9`. We need to compare the numeric values of each version component.

Reply:
Ah, the one edge case I didn't test. I still think it could be simplified, there's no need to pad with zeros as you could just compare the minimum segments available. Maybe something like this:
```JS
fileNames.sort((a, b) => {
    a = a.split('.');
    b = b.split('.');

    const length = Math.min(a.length, b.length);
    for (let i = 0; i < length; i++) {

        // Compare each segment, invalid segments are sent to back.
        const delta = (parseInt(b[i], 10) || -1) - (parseInt(a[i], 10) || -1);
        if (delta) {
            return delta;
        }
    }
    return -1;
});
```

Reply:
How about this?
```javascript
fileNames.sort((text1, text2) => {
	// Flip result to sort in descending order. (ie: Highest version is first.)
	return versionStringComparer(text1, text2) * (-1);
});

// ...

function versionStringComparer(text1, text2) {
	// Split strings into version component arrays. Example: '1.2.3' -> ['1', '2', '3']
	const array1 = text1.split('.');
	const array2 = text2.split('.');

	// Compare the 2 given strings by their numeric components.
	// If they match numerically, then do a string comparison.
	const maxLength = Math.max(array1.length, array2.length);
	for (let index = 0; index < maxLength; index++) {
		const value1 = (index < array1.length) ? (Number.parseInt(array1[index], 10) || 0) : 0;
		const value2 = (index < array2.length) ? (Number.parseInt(array2[index], 10) || 0) : 0;
		const delta = value1 - value2;
		if (delta !== 0) {
			return delta;
		}
	}
	return text1.localeCompare(text2);
}
```

Reply:
@jquick-axway Lets go with that 👍